### PR TITLE
AutoStandby: fix (re)scheduling

### PR DESCRIPTION
**Pull request by zwim**
I had the feeling that `autostandby` has Alzheimer. Sometimes my Sage forgot to standby. 

@NiLuJe : It's not the metal table, it's the time of the day. When reading at night I have more time and sometimes `autodim` or `autowarmth` knock in. If they knock in within the `wake_in >=3` is false, autostandby looses its scheduling. (Autodim is a real stress-test.)

On normal reading (with no autodim starting, during the day when warmth does not change) you can sometimes provoke the failure if you tap right before an automatic footer update (around second 57, 58).

This PR should fix the problem.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9258)

### CODE REVIEW

**Comment by Frenzie** [comment]
Also I think it doesn't work anymore while charging?

**Comment by NiLuJe** [comment]
> Also I think it doesn't work anymore while charging?

That's on purpose ^^. (You can't enter PM states while charging on < Mk.7 devices anyway).

File: plugins/autosuspend.koplugin/main.lua [hunk 1]
```diff
@@ -606,6 +606,8 @@ function AutoSuspend:AllowStandbyHandler()
         -- This shouldn't prevent us from actually consuming any pending input events first,
         -- because if we were woken up by user input, those events should already be in the evdev queue...
         UIManager:consumeInputEarlyAfterPM(true)
+    else
+        self:_start_standby()
```
Comment (NiLuJe) [inline plugins/autosuspend.koplugin/main.lua:610]: I'd possibly guard it behind an `if not self.going_to_suspend` branch to mimic `onLeaveStandby`, but that looks sensible, yeah ;).

**Comment by Frenzie** [comment]
> That's on purpose ^^. (You can't enter PM states while charging on < Mk.7 devices anyway).

Well, it used to show a standby screen and turn off the light while charging anyway. Whether it was actually standby is fairly immaterial in that context since it has a supply of energy but the touch screen response and light previously seemed as expected anyway and now they don't. ;-)

**Comment by Frenzie** [comment]
Basically it's only some 15 minutes that my cat can mess up everything as opposed to 15 hours.

**Comment by NiLuJe** [comment]
Spoiler alert: the shutdown timer is *also* delayed ;p.

Too late/early to confirm, but it might make sense to not do that once *charged*, though.

(Sidebar: you're talking about suspend & shutdown, not standby, right?)

**Comment by Frenzie** [comment]
> (Sidebar: you're talking about suspend & shutdown, not standby, right?)

I'm talking about the change in behavior, not about any technicalities.

**Comment by NiLuJe** [comment]
Except the technicality defines the behavior ;).

While trying to standby while charging makes absolutely no sense, regardless of the device's capabilities, the same may not be said of suspend & shutdown. (It, err, certainly makes no sense for shutdown, at the very least). 

I'll double-check tonight, because I can't quite recall what it did before (note to self: probably related to the new fully charged status).

**Comment by NiLuJe** [comment]
> I'll double-check tonight, because I can't quite recall what it did before (note to self: probably related to the new fully charged status). 

Yup, that's it. Will tweak tonight ;).

**Comment by Frenzie** [comment]
If it was in fake standby/suspend because real standby/suspend wasn't possible until later, then that is the desired behavior.

**Comment by Frenzie** [comment]
> Note that it did so only when fully charged, and I intend to keep that distinction ;).

It's certainly possible that I never noticed the distinction. Charging doesn't take hours after all.

**Pull request merged**